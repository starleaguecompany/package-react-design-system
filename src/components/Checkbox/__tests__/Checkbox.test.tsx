import * as React from 'react'
import { render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'

import { Checkbox } from '../../../index'

describe('Checkbox', () => {
  test('should render correctly with no props', () => {
    render(<Checkbox />)

    expect(screen.getByTestId('Checkbox')).toMatchSnapshot()
    expect(screen.getByTestId('Checkbox')).toHaveClass('container', { exact: true })
    expect(screen.getByRole('checkbox')).toHaveClass('input', { exact: true })
  })

  test('should render correctly with attributes', () => {
    render(<Checkbox id="test-id" className="test-class" />)

    expect(screen.getByTestId('Checkbox')).toMatchSnapshot()
    expect(screen.getByTestId('Checkbox')).toHaveClass('test-class')
    expect(screen.getByRole('checkbox')).toHaveAttribute('id', 'test-id')
  })

  test('should render correctly with label', () => {
    render(<Checkbox>Testing checkbox</Checkbox>)

    expect(screen.getByTestId('Checkbox')).toMatchSnapshot()
    expect(screen.getByText('Testing checkbox')).toBeVisible()
    expect(screen.getByText('Testing checkbox')).toHaveClass('label')
  })

  test('should render correctly with checked prop', () => {
    render(<Checkbox checked />)

    expect(screen.getByTestId('Checkbox')).toMatchSnapshot()
    expect(screen.getByRole('checkbox')).toBeChecked()
  })

  test('should render correctly with disabled prop', () => {
    render(<Checkbox disabled />)

    expect(screen.getByTestId('Checkbox')).toMatchSnapshot()
    expect(screen.getByRole('checkbox')).toBeDisabled()
  })

  test('should render correctly with name prop', () => {
    render(<Checkbox name="Testing checkbox" />)

    expect(screen.getByTestId('Checkbox')).toMatchSnapshot()
    expect(screen.getByRole('checkbox')).toHaveAttribute('name', 'Testing checkbox')
  })

  test('should render correctly with value prop', () => {
    render(<Checkbox value="Testing checkbox" />)

    expect(screen.getByTestId('Checkbox')).toMatchSnapshot()
    expect(screen.getByRole('checkbox')).toHaveAttribute('value', 'Testing checkbox')
  })

  test('should render correctly with readonly prop', () => {
    render(<Checkbox readOnly />)

    expect(screen.getByTestId('Checkbox')).toMatchSnapshot()
    expect(screen.getByTestId('Checkbox')).toHaveClass('readOnly')
  })

  test('should execute onChange callback on checking', () => {
    const onChangeCallback = jest.fn()
    render(<Checkbox onChange={onChangeCallback} />)

    expect(screen.getByTestId('Checkbox')).toMatchSnapshot()
    userEvent.click(screen.getByRole('checkbox'))
    expect(onChangeCallback).toBeCalledTimes(1)
  })

  test('should render correctly on external value update', () => {
    const onChangeCallback = jest.fn()
    const { rerender } = render(<Checkbox checked={false} onChange={onChangeCallback} />)

    expect(screen.getByTestId('Checkbox')).toMatchSnapshot()
    rerender(<Checkbox checked={true} onChange={onChangeCallback} />)
    expect(screen.getByRole('checkbox')).toBeChecked()
    expect(onChangeCallback).toBeCalledTimes(0)
  })
})
