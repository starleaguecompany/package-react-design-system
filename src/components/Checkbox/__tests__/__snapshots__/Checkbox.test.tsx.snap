// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox should execute onChange callback on checking 1`] = `
<label
  class="container"
  data-qa="Checkbox"
>
  <input
    class="input"
    type="checkbox"
    value=""
  />
  <div
    class="control"
  />
</label>
`;

exports[`Checkbox should render correctly on external value update 1`] = `
<label
  class="container"
  data-qa="Checkbox"
>
  <input
    class="input"
    type="checkbox"
    value=""
  />
  <div
    class="control"
  />
</label>
`;

exports[`Checkbox should render correctly with attributes 1`] = `
<label
  class="test-class container"
  data-qa="Checkbox"
>
  <input
    class="input"
    id="test-id"
    type="checkbox"
    value=""
  />
  <div
    class="control"
  />
</label>
`;

exports[`Checkbox should render correctly with checked prop 1`] = `
<label
  class="container checked"
  data-qa="Checkbox"
>
  <input
    checked=""
    class="input"
    type="checkbox"
    value=""
  />
  <div
    class="control checked"
  >
    <svg
      class="icon"
      data-qa="Check"
      fill="none"
      viewBox="0 0 20 20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="m6 11 3 3 6-7"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="1.5"
      />
    </svg>
  </div>
</label>
`;

exports[`Checkbox should render correctly with disabled prop 1`] = `
<label
  class="container disabled"
  data-qa="Checkbox"
>
  <input
    class="input"
    disabled=""
    type="checkbox"
    value=""
  />
  <div
    class="control disabled"
  />
</label>
`;

exports[`Checkbox should render correctly with label 1`] = `
<label
  class="container"
  data-qa="Checkbox"
>
  <input
    class="input"
    type="checkbox"
    value=""
  />
  <div
    class="control"
  />
  <span
    class="label"
  >
    Testing checkbox
  </span>
</label>
`;

exports[`Checkbox should render correctly with name prop 1`] = `
<label
  class="container"
  data-qa="Checkbox"
>
  <input
    class="input"
    name="Testing checkbox"
    type="checkbox"
    value=""
  />
  <div
    class="control"
  />
</label>
`;

exports[`Checkbox should render correctly with no props 1`] = `
<label
  class="container"
  data-qa="Checkbox"
>
  <input
    class="input"
    type="checkbox"
    value=""
  />
  <div
    class="control"
  />
</label>
`;

exports[`Checkbox should render correctly with readonly prop 1`] = `
<label
  class="container readOnly"
  data-qa="Checkbox"
>
  <input
    class="input"
    type="checkbox"
    value=""
  />
  <div
    class="control"
  />
</label>
`;

exports[`Checkbox should render correctly with value prop 1`] = `
<label
  class="container"
  data-qa="Checkbox"
>
  <input
    class="input"
    type="checkbox"
    value="Testing checkbox"
  />
  <div
    class="control"
  />
</label>
`;
